File: solutions/204. Count Primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if(n < 2){
            return 0;
        }
        
        vector<bool> primee(n + 1) ;
        for(long long int i = 2; i < n + 1; i++){
            if(!primee[i]){
                for(long long int j = i * i; j < n + 1; j += i){
                    primee[j] = true;
                }
            }
        }
        
        int ans = 0;
        for(int i = 2; i < n ; i++){
            if(primee[i] == false){
                ans++;
            }
        }
        
        return ans;
    }
};

File: solutions/204. Count Primes.java
class Solution {
    public int countPrimes(int n) {
        boolean[] isPrime = new boolean[n];
        Arrays.fill(isPrime, true);
        for (int i = 2; i * i < n; i++) {
            if(isPrime[i]==false) continue;
            else {
                for (int j = i * i; j < n; j +=i) {
                    isPrime[j] = false;
                }
            }
        }
        int count = 0;
        for (boolean x : isPrime) {
            if (x==true) count++;
        }
        return count-2;
    }
}
